### DISCUSSION TITLE
Does the java wrapper have a similar rdkit.Chem.QED module in the python API

### DISCUSSION THREAD

**Discussion by miracle1111111** [rdkit/rdkit#3841] [Q&A] [answered]
I need rdkit.Chem.QED.QEDproperties (MW, ALOGP, HBA, HBD, PSA, ROTB, AROM, ALERTS) in python. Then there is no quantitative estimate of drug similarity in the java wrapper. Where is it located?
Thank you very much！

**Comment by greglandrum** [comment, marked as answer]
The RDKit's implementation of QED is currently written in Python. In order for it to be available in the Java wrappers someone would need to either do a separate Java implementation or, better, a C++ implementation which we could then use to also replace the Python version.

**Reply by miracle1111111** [reply]
Well, I need QED implemented in java, I don't have any idea about it. Excuse me, how should I achieve it

**Reply by greglandrum** [reply]
If you can't find an open-source implementation in Java that you can use, then you'll probably need to implement it yourself.

The paper describing the method is here: https://doi.org/10.1038/nchem.1243

Here's the RDKit Python source code, which might be helpful:
https://github.com/rdkit/rdkit/blob/master/rdkit/Chem/QED.py

**Reply by miracle1111111** [reply]
Thanks, this is very helpful to me!
